### ISSUE TITLE
 [JAVA] code generation of test classes generate invalid code when having json in subfolder

### ISSUE THREAD

**Issue by sdkrum** [swagger-api/swagger-codegen#9221]
##### Description

Works nice so far, at least for the classes in main/java, however the classes in test/java do not compile. I checked a bit, and it looks like the class names generated in the test classes gets wrong, if the json file defining this class resides in a sub-directory of the main swagger file. In this case, it is TerminKoder enum, which is placed in a file [common/Terminkoder.json](https://fsapi-test.uio.no/models/common/Terminkoder.json). The generated java class is nicely placed in the model folder, but in the test classes, they assume the class would be named CommonTerminkoderJson, instead for Terminkoder.

Interesting enough, that happens for this [Token class](https://fsapi-test.uio.no/models/Token.json) too, and it is not in a sub-directory

##### Swagger-codegen version

I used a version checked out from the v3.0.5 tag 
##### Swagger declaration file content or url
[https://fsapi-test.uio.no/models/swagger.json](https://fsapi-test.uio.no/models/swagger.json)

##### Command line used for generation
java -jar swagger-codegen-cli.jar generate -i https://fsapi-test.uio.no/models/swagger.json -l java -c java-config.json -o client

with a java-config.json like:
```json
{
  "modelPackage" : "no.unit.swagger.fsapi.client.model",
  "apiPackage"   : "no.unit.swagger.fsapi.client.api",
  "invokerPackage" : "no.unit.swagger.fsapi.client",
  "groupId" : "no.unit.swagger.fsapi",
  "artifactId"  : "model",
  "artifact-version" : "0.0.1-SNAPSHOT",
  "fullJavaUtil" : true,
  "dateLibrary" : "java8",
  "library" :  "okhttp-gson"
}
```

##### Steps to reproduce

1. Run the code generator as described
2. change in the directory client `cd client`
3. run maven `mvn clean package`

Results in a compile error of the test classes

##### Related issues/PRs

Did not find something which seems directly related to this here

##### Suggest a fix/enhancement

Basically, i have no idea, i just started using this tool, and it took a while to make it run at all. I tried with the versions from the maven repositories first, and they failed for different reasons to generate any code, so i choosed to check out the source code and try with this tag, as it seems the lates.

**Comment by sdkrum** [comment]
Forget about that, after cleaning out the target directory and running it again it worked nicely!
